docs/en/contributor/Building-Alluxio-From-Source.md

```diff
@@ -52,6 +42,28 @@ $ git tag
 $ git checkout <TAG_NAME>
 ```
 
+## (Optional) Checkout Building Environment Using Docker
+
+This section guides you to setup pre-configured compilation environment based on our published docker image.
+You can skip this section and build Alluxio source code if JDK and Maven are already installed locally.
+
+Start a container named `alluxio-build` based on this image and get into this container to proceed:
+
+```console
+$ docker run -itd \
+  --network=host \
+  -v ${ALLUXIO_HOME}:/alluxio  \
+  -v ${HOME}/.m2:/root/.m2 \
+  --name alluxio-build \
+  alluxio/alluxio-maven bash
+
+$ docker exec -it -w /alluxio alluxio-build bash
+```
+
+Note that,
+- Container path `/alluxio` is mapped to host path `${ALLUXIO_HOME}`, so the binary built will still be accessible outside the container afterwards.
+- Container path `/root/.m2` is mapped to host path `${HOME}/.m2` to leverage your local copy of the maven cache. This is optional.
+
```

Comment:
When done using the container, destroy it by running `docker rm -f alluxio-build`

Reply:
added